backend/database.py

```
@@ -811,56 +892,159 @@ def cleanup_test_data() -> Dict[str, int]:
+    try:
+        cursor.executemany('''
+            INSERT OR REPLACE INTO graph_nodes (id, type, label, metadata)
+            VALUES (:id, :type, :label, :metadata)
+        ''', nodes)
+
+        cursor.executemany('''
+            INSERT OR REPLACE INTO graph_edges (source_id, target_id, weight, relation_type)
+            VALUES (:source_id, :target_id, :weight, :relation_type)
+        ''', edges)
+        conn.commit()
+    except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)